### ISSUE TITLE
Closed UIs are not detected reliably

### ISSUE THREAD

**Issue by vaadin-bot** [vaadin/framework#5461]
**Originally by _Gast B_**

___
I have upgraded vaadin from v7.2.4 to 7.2.5 and now i see the problem, that closed UIs are no longer detected "immediately" during run time, but only if another user opens a new session. Furthermore, calling UI.close() is suspended until next user logs in. However, calling UI.accessSynchroniously() still works (which i use for a workaround to free resources).

Scenario:
* Servlet with VaadinServletConfiguration annotation with heartbeat parameter set to 30.
* UI parameter is set to UI.class.
* Correct (custom) UI is determined through UIProvider dynamically attached to an incoming VaadinSession.
* "detach" method overwritten and DetachListener appended to Custom UI
* On detach a custom Event has to be send to listeners to inform them on the closed UI
* Open a browser window (here firefox 31.0 and Safari 7.0.4) and surf to servlet
* Close the browser window
----------
* Nothing happens
* The UI is not closed nor detached
* UI.detach() and UI.close() methods are not called
* DetachListener is not called 
* VaadinSessionDestroyListener on VaadinServlet is NOT triggered
-----------
* Open a new browser window and surf to the servlet
* Old Session gets properly closed/destroyed

___
 Imported from https://dev.vaadin.com/ issue #14281
